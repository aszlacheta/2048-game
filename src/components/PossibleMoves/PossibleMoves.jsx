import { useTranslation } from 'react-i18next';
import './PossibleMoves.scss';

/**
 * Component used to represent number of possible moves for given game
 * @param {{number: number, maxMoves: number}} props React props that represent:
 *      1) number - number of possible moves that left for given game 
 *      2) maxMoves - maximum number of moves
 * @returns {ReactElement}
 */
export default function PossibleMoves({ number, maxMoves }) {
    const { t } = useTranslation();
    const width = number > 0 ? (number / maxMoves) : 0;

    return (
        <div className="possible-moves" title={t('possibleMoves.label', { number })}>
            <div className="possible-moves-number">{number}</div>
            <div className="possible-moves-progress" style={{ "--width": width }} />
        </div>
    )
}