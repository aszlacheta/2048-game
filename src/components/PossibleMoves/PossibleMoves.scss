.possible-moves {
    display: flex;
}

.possible-moves-progress {
    position: relative;
    height: 30px;
    width: 80%;
    background: linear-gradient(180deg, #000000, #0e0e0e, #171717, #1f1f1f, #272727, #303030, #383838, #414141);
    border-radius: 15px;

    &:after {
        content: "";
        position: absolute;
        top: 0;
        left: 0;
        width: calc(var(--width) * 100%);
        height: 100%;
        background: linear-gradient(180deg, #ffd500, #ffcb12, #ffc11f, #ffb728, #ffae31, #ffa538, #ff9c3f, #fb9445);
        border-radius: 14px;
    }
}

.possible-moves-number {
    width: 30px;
    height: 30px;
    margin-right: 0.5rem;
    background: linear-gradient(180deg, #ffd500, #ffcb12, #ffc11f, #ffb728, #ffae31, #ffa538, #ff9c3f, #fb9445);
    border-radius: 15px;
    text-align: center;
    line-height: 28px;
}
