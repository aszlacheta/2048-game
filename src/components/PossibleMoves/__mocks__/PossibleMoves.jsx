import React from 'react';

const PossibleMoves = () => <div className="mocked-possible-moves" />

module.exports = PossibleMoves;