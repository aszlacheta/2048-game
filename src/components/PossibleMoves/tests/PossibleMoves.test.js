import renderer from 'react-test-renderer';
import PossibleMoves from '../PossibleMoves';

describe('Possible moves component', () => {
    it('should render properly without props', () => {
        const component = renderer.create(<PossibleMoves />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render properly with props provided', () => {
        const component = renderer.create(<PossibleMoves number={3} maxMoves={6}/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render properly with wrong props provided', () => {
        const component = renderer.create(<PossibleMoves number={'a'} maxMoves={'b'}/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});