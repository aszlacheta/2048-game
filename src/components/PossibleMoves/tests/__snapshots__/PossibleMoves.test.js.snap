// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Possible moves component should render properly with props provided 1`] = `
<div
  className="possible-moves"
  title="possibleMoves.label"
>
  <div
    className="possible-moves-number"
  >
    3
  </div>
  <div
    className="possible-moves-progress"
    style={
      Object {
        "--width": 0.5,
      }
    }
  />
</div>
`;

exports[`Possible moves component should render properly with wrong props provided 1`] = `
<div
  className="possible-moves"
  title="possibleMoves.label"
>
  <div
    className="possible-moves-number"
  >
    a
  </div>
  <div
    className="possible-moves-progress"
    style={
      Object {
        "--width": 0,
      }
    }
  />
</div>
`;

exports[`Possible moves component should render properly without props 1`] = `
<div
  className="possible-moves"
  title="possibleMoves.label"
>
  <div
    className="possible-moves-number"
  />
  <div
    className="possible-moves-progress"
    style={
      Object {
        "--width": 0,
      }
    }
  />
</div>
`;
